### PR TITLE
Allow leading underscores in idents

### PR DESCRIPTION
**Pull request by calder**
# Summary

`_` is not a member of `XID_Start` because some languages don't consider it a valid leading character (certain SQL dialects for example). In practice, leading `_`'s are accepted by most languages and even have a special role in many modern languages where they indicate intentionally ignored variables. Accepting them in `ident()` should make it more useful and less surprising.

# References

* https://www.unicode.org/reports/tr31/
* https://www.unicode.org/reports/tr31/#NFKC_Modifications

### CODE REVIEW

File: src/text.rs [hunk 1]
```diff
@@ -528,6 +528,33 @@ pub mod unicode {
...
+        fn test_err(self: Self, input: &'a str) {
+            assert_eq!(
+                self.parse(input),
+                ParseResult {
+                    output: None,
+                    errs: vec![EmptyErr::default()]
+                }
+            );
+        }
+    }
```
Comment (calder) [inline src/text.rs:556]: Let me know if:
* This trait seems like overkill and you'd rather make these free functions.
* This trait seems generally useful and you'd rather move them to a `cfg(test)` block in `util.rs`.

Comment (zesterer) [reply src/text.rs:556]: Hmm, I do think this is overkill. Free functions would be a little simpler.

**Review by zesterer — CHANGES_REQUESTED** [review]
Thanks so much for the PR! I actually noticed this just a few days ago when playing around on a personal project, but didn't stop to check whether XID_Start actually permitted underscores.

File: src/text.rs (hunk 1 above)
Comment (calder) [reply src/text.rs]: Done.

**Comment by zesterer** [comment]
Thanks again!

**Pull request merged**